templates/default/condaexport.sh.erb

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# conda list [-c $1]
+
+channel=
+if [[ $# -ne 5 ]]; then
+  echo "incorrect usage"
+  echo "exportdir project host environment user"
+  exit 1
+fi
+
+CONDA_USER=<%= node['conda']['user'] %>
+GLASSFISH_USER=<%= node['glassfish']['user'] %>
+
+EXPORTDIR=$1
+PROJECT=$2
+HOST=$3
+ENVIRONMENTFILE=$4
+IMPERSONATED_USERNAME=$5
+
+function finish {
+    rm -rf $EXPORTDIR
+}
+
+trap finish EXIT
+
+cd $EXPORTDIR
+
+HOSTNAME=$(hostname)
+
+if [[ "${HOST}" == "${HOSTNAME}" ]]; then
+  su ${CONDA_USER} -c "<%= node['conda']['base_dir'] %>/bin/conda env export -n ${PROJECT}" > ${EXPORTDIR}/${ENVIRONMENTFILE}
+else
+  su ${GLASSFISH_USER} -c "ssh ${CONDA_USER}@${HOST} <%= node['conda']['base_dir'] %>/bin/conda env export -n ${PROJECT}" > ${EXPORTDIR}/${ENVIRONMENTFILE}
+fi
+
```

Comment:
You need to check if the ssh command (or even the conda env export command) failed or not. Otherwise, you'll print an error message into the environmentfile and upload it. 
Something like:

if [ $? -ne 0 ] ; then
   exit 2
fi